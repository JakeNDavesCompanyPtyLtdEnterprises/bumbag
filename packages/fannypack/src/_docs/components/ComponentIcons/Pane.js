import React from 'react';

const SvgComponent = props => (
  <svg width={84} height={37} fill="none" {...props}>
    <rect x={0.5} y={0.5} width={83} height={36} rx={3.5} fill="#fff" stroke="#212121" />
  </svg>
);

export default SvgComponent;
