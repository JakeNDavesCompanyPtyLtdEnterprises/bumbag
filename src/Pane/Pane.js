// @flow
import React, { type Node } from 'react';

import _Pane from './styled';

type Props = {
  border: boolean | 'shadow',
  className?: string,
  children?: Node,
  isFullWidth?: boolean
};

const Pane = ({ children, ...props }: Props) => <_Pane {...props}>{children}</_Pane>;

Pane.defaultProps = {
  className: undefined,
  children: undefined,
  isFullWidth: false
};

export default Pane;
