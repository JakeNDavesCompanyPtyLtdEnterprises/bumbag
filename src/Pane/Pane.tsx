import * as React from 'react';
// @ts-ignore
import PropTypes from 'prop-types';
import { BoxProps as ReakitBoxProps } from 'reakit/ts';
import { Omit } from '../types';

import _Pane from './styled';

export interface LocalPaneProps {
  border?: boolean | 'shadow';
  className?: string;
  children?: React.ReactNode;
  isFullWidth?: boolean;
}
export type PaneProps = LocalPaneProps & Omit<ReakitBoxProps, 'border'>;

export const Pane: React.FunctionComponent<LocalPaneProps> = ({ border, children, ...props }) => (
  <_Pane styledBorder={border} {...props}>
    {children}
  </_Pane>
);

Pane.propTypes = {
  border: PropTypes.oneOfType([PropTypes.bool, PropTypes.string, PropTypes.oneOf(['shadow'])]),
  className: PropTypes.string,
  children: PropTypes.node,
  isFullWidth: PropTypes.bool
};
Pane.defaultProps = {
  border: false,
  className: undefined,
  children: undefined,
  isFullWidth: false
};

export default Pane as React.FunctionComponent<PaneProps>;
