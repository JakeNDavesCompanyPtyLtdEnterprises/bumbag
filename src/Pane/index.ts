export { default } from './Pane';
export { default as Pane } from './Pane';
