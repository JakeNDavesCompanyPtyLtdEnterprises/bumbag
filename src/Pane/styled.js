import { palette, theme } from 'styled-tools';
import styled, { css } from '../styled';
import { Box } from '../primitives';

const Pane = styled(Box)`
  display: inline-flex;
  border-radius: 3px;
  flex-wrap: wrap;

  ${props =>
    props.border === 'shadow' &&
    css`
      box-shadow: 0px 2px 4px 0px ${palette('whiteDarkest')}, 0px 0px 0px 1px ${palette('whiteDarkest')};

      & {
        ${theme('fannypack.Pane.border.shadow')};
      }
    `};
  ${props =>
    props.border === true &&
    css`
      border: 1px solid ${palette('whiteDarkest')};

      & {
        ${theme('fannypack.Pane.border.default')};
      }
    `};
  ${props =>
    props.isFullWidth &&
    css`
      width: 100%;
    `};

  & {
    ${theme('fannypack.Pane.base')};
  }
`;

export default Pane;
