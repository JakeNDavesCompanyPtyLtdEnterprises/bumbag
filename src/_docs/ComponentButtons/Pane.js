import React from 'react';

const SvgComponent = props => (
  <svg width={84} height={37} fill="none" {...props}>
    <rect x={0.5} y={0.5} width={83} height={36} rx={3.5} stroke="#212121" />
    <rect x={12} y={9} width={28} height={4} rx={2} fill="#3926A5" />
    <rect x={12} y={17} width={60} height={4} rx={2} fill="#3926A5" />
    <rect x={12} y={25} width={60} height={4} rx={2} fill="#3926A5" />
  </svg>
);

export default SvgComponent;
