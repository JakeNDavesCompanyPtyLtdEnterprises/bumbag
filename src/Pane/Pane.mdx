---
name: Pane
route: /layout/pane
menu: Layout
---

import { Playground, PropsTable } from 'docz';
import Pane from '../Pane';

# Pane

## Import

`import { Pane } from 'fannypack'`

## Basic Usage

A `<Pane>` is the same as a `<Box>`, except it has a default border radius, and has the ability to add different types of borders to it. Using `<Pane>` and `<Box>` together are most likely the two building blocks for the layout of your app.

<Playground>
  <Pane backgroundColor="whiteDarker" padding="large">Hello world</Pane>
</Playground>

## Shadowed

<Playground>
  <Pane border="shadow" padding="large">Hello world</Pane>
</Playground>

## Bordered

<Playground>
  <Pane border padding="large">Hello world</Pane>
</Playground>

## Full width

<Playground>
  <Pane border isFullWidth padding="large">Hello world</Pane>
</Playground>

## Props

<PropsTable of={Pane} />

## Theming

### Schema

```jsx
{
  base: string | Object,
  border: {
    default: string | Object,
    shadow: string | Object
  }
}
```
