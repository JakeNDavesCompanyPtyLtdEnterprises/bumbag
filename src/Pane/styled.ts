import { palette, theme } from 'styled-tools';

import styled, { css } from '../styled';
import { Box } from '../primitives';
import { PaneProps } from './Pane';

const Pane = styled(Box)<PaneProps & { styledBorder?: boolean | 'shadow' }>`
  background-color: white;
  border-radius: 3px;
  display: inline-flex;
  flex-wrap: wrap;

  ${props =>
    props.styledBorder === 'shadow' &&
    css`
      box-shadow: 0px 2px 4px 0px ${palette('whiteDarkest')}, 0px 0px 0px 1px ${palette('whiteDarkest')};

      & {
        ${theme('fannypack.Pane.border.shadow')};
      }
    `};
  ${props =>
    props.styledBorder === true &&
    css`
      border: 1px solid ${palette('whiteDarkest')};

      & {
        ${theme('fannypack.Pane.border.default')};
      }
    `};
  ${props =>
    props.isFullWidth &&
    css`
      width: 100%;
    `};

  & {
    ${theme('fannypack.Pane.base')};
  }
`;

export default Pane;
