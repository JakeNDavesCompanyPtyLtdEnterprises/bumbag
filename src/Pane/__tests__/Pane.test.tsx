import * as React from 'react';
import render from '../../_utils/tests/render';
import Pane from '../Pane';

it('renders correctly for a basic pane', () => {
  const { container } = render(<Pane>test</Pane>);
  expect(container.firstChild).toMatchSnapshot();
});

it('renders correctly for a pane with a border', () => {
  const { container } = render(<Pane border>test</Pane>);
  expect(container.firstChild).toMatchSnapshot();
});

it('renders correctly for a pane with a shadow', () => {
  const { container } = render(<Pane border="shadow">test</Pane>);
  expect(container.firstChild).toMatchSnapshot();
});

it('renders correctly for a full width pane', () => {
  const { container } = render(<Pane isFullWidth>test</Pane>);
  expect(container.firstChild).toMatchSnapshot();
});
