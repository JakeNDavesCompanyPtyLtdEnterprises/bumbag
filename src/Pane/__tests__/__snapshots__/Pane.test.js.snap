// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly for a basic pane 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c0 {
  box-sizing: border-box;
  font-family: system-ui,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Fira Sans', 'Droid Sans','Helvetica Neue',sans-serif;
  font-size: 16px;
  font-weight: 400;
  line-height: 1.5;
  margin: 0;
  padding: 0;
  overflow-x: hidden;
  -webkit-font-smoothing: antialiased;
  text-rendering: optimizeLegibility;
  color: #435a6f;
}

.c0 *,
.c0 *::before,
.c0 *::after {
  box-sizing: inherit;
}

.c0 *:focus {
  outline: 2px solid #7467c0;
  outline-offset: 2px;
}

.c2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  border-radius: 3px;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

<div
  class="c0 c1"
>
  <div
    class="c2 c1"
  >
    test
  </div>
</div>
`;

exports[`renders correctly for a full width pane 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c0 {
  box-sizing: border-box;
  font-family: system-ui,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Fira Sans', 'Droid Sans','Helvetica Neue',sans-serif;
  font-size: 16px;
  font-weight: 400;
  line-height: 1.5;
  margin: 0;
  padding: 0;
  overflow-x: hidden;
  -webkit-font-smoothing: antialiased;
  text-rendering: optimizeLegibility;
  color: #435a6f;
}

.c0 *,
.c0 *::before,
.c0 *::after {
  box-sizing: inherit;
}

.c0 *:focus {
  outline: 2px solid #7467c0;
  outline-offset: 2px;
}

.c2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  border-radius: 3px;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
}

<div
  class="c0 c1"
>
  <div
    class="c2 c1"
  >
    test
  </div>
</div>
`;

exports[`renders correctly for a pane with a border 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c0 {
  box-sizing: border-box;
  font-family: system-ui,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Fira Sans', 'Droid Sans','Helvetica Neue',sans-serif;
  font-size: 16px;
  font-weight: 400;
  line-height: 1.5;
  margin: 0;
  padding: 0;
  overflow-x: hidden;
  -webkit-font-smoothing: antialiased;
  text-rendering: optimizeLegibility;
  color: #435a6f;
}

.c0 *,
.c0 *::before,
.c0 *::after {
  box-sizing: inherit;
}

.c0 *:focus {
  outline: 2px solid #7467c0;
  outline-offset: 2px;
}

.c2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  border-radius: 3px;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  border: 1px solid #e6e6e6;
}

<div
  class="c0 c1"
>
  <div
    class="c2 c1"
  >
    test
  </div>
</div>
`;

exports[`renders correctly for a pane with a shadow 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c0 {
  box-sizing: border-box;
  font-family: system-ui,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Fira Sans', 'Droid Sans','Helvetica Neue',sans-serif;
  font-size: 16px;
  font-weight: 400;
  line-height: 1.5;
  margin: 0;
  padding: 0;
  overflow-x: hidden;
  -webkit-font-smoothing: antialiased;
  text-rendering: optimizeLegibility;
  color: #435a6f;
}

.c0 *,
.c0 *::before,
.c0 *::after {
  box-sizing: inherit;
}

.c0 *:focus {
  outline: 2px solid #7467c0;
  outline-offset: 2px;
}

.c2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  border-radius: 3px;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  box-shadow: 0px 2px 4px 0px #e6e6e6,0px 0px 0px 1px #e6e6e6;
}

<div
  class="c0 c1"
>
  <div
    class="c2 c1"
  >
    test
  </div>
</div>
`;
